### ISSUE TITLE
Add the new Scenario features to CustomizeScenarioDialog

### ISSUE THREAD

**Issue by AaronGullickson** [MegaMek/mekhq#3927]
As part of story arcs development, I have built a lot of new capabilities into the basic Scenario object. I also migrated some stuff from AtBScenario to basic Scenario. So you can now have the following in basic Scenario:

* Pre-specified board
* Pre-specified deployment.
* Pre-specified scenario objectives (ScenarioObjective).
* Pre-specified opponents and allies (BotForce). 
* Pre-specified planetary conditions.

However, it is not yet possible to access these capabilities from within MekHQ. I should add these new capabilities to CustomizeScenarioDialog. This is important prepatory work toward building a Story Arc editor. However, it can also be useful for players more generally because they can pre-specify all of this stuff before starting the scenario so that when they start it, they are just ready to go.

**Comment by IllianiCBT** [comment]
I don’t know if #3779 is still a needed function, and if so, whether it could be covered by this?

**Comment by Thom293** [comment]
Oh man these are awesome.  So many ways to use this.  I cant wait to see how this works.

**Comment by AaronGullickson** [comment]
@IllianiCBT I have been thinking about the date issue as well, but I think thats a separate thing. Currently, the date is left empty until added after the scenario is completed, which makes sense. However, I think it would be useful to have the date pre-specified to prevent advancing the day until scenarios with the current date are completed.

**Comment by AaronGullickson** [comment]
Ok, I think it makes sense to add a "Change Date" to the dialog as well, which I think would obviate the need for that PR.

I have another idea about preventing user from advancing days when they have scenarios "due" but that is a separate issue.
